extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufReader.java

```diff
@@ -217,8 +225,10 @@ private void addPointRows(Point point, Metric metric, LabelContext labelContext)
   }
 
   @Override
-  public CloseableIterator<InputRowListPlusRawValues> sample()
+  public CloseableIterator<InputRowListPlusRawValues> sample() throws IOException
   {
-    return read().map(row -> InputRowListPlusRawValues.of(row, ((MapBasedInputRow) row).getEvent()));
+    try (CloseableIterator<InputRow> iterator = read()) {
+      return iterator.map(row -> InputRowListPlusRawValues.of(row, ((MapBasedInputRow) row).getEvent()));
+    }
```

Comment:
are we sure we should use try-with-resources here? If we close the underlying iterator before returning the mapped one, wouldn't this cause issues? It seems it should be the responsibility of the called to close the returned iterator, not this method.

Reply:
You're right. Only that [InputEntityIteratingReader](https://github.com/confluentinc/druid/blob/0.22.1-confluent/core/src/main/java/org/apache/druid/data/input/impl/InputEntityIteratingReader.java#L87-L99) which is one of the few users of this call doesn't close it as it should when it handles the `IOException`. 

I guess it doesn't blow up because of low coverage of sample methods (I found out manually on the UI). Reverting to unblock this for now and we can reconsider in the future.

Reply:
Btw, in the Hybrid class, IOException has to stay because we are calling the interface method (that has it)

Reply:
good catch about [InputEntityIteratingReader](https://github.com/confluentinc/druid/blob/0.22.1-confluent/core/src/main/java/org/apache/druid/data/input/impl/InputEntityIteratingReader.java#L87-L99) maybe you can file an issue upstream to raise that potential problem